addons/website_mail/static/src/js/follow.js

```diff
@@ -4,25 +4,41 @@ odoo.define('website_mail.follow', function (require) {
 var publicWidget = require('web.public.widget');
 
 publicWidget.registry.follow = publicWidget.Widget.extend({
-    selector: '.js_follow',
+    selector: '#wrapwrap:has(.js_follow)',
     disabledInEditableMode: false,
 
+    /**
+     * @override
+     */
     start: function () {
         var self = this;
         this.is_user = false;
+        var $jsFollowEls = this.$el.find('.js_follow');
 
         var always = function (data) {
-            self.is_user = data.is_user;
-            self.email = data.email;
-            self.toggle_subscription(data.is_follower, data.email);
-            self.$target.removeClass('d-none');
+            self.is_user = data[0].is_user;
+            var $jsFollowToEnable = $();
+            _.each(data[1], (resIds, model) => _.each(resIds, (resId) => {
+                $jsFollowToEnable = $jsFollowToEnable.add(
+                    $jsFollowEls.filter('[data-id="' + resId + '"][data-object="' + model + '"]')
+                );
+            }));
+            self._enable_subscription(data[0].email, $jsFollowToEnable);
+            self._disable_subscription(data[0].email, $jsFollowEls.not($jsFollowToEnable));
+            $jsFollowEls.removeClass('d-none');
         };
 
+        var records = _.chain(_.map($jsFollowEls, function ($el) {
+                return {'model': $($el).data('object'), 'res_id': $($el).data('id')};
+            }))
+            .groupBy('model')
+            .mapObject(group => _.pluck(group, 'res_id'))
+            .value();
```

Comment:
- Don't name HTMLElement variables with a `$`; if you write `$($` it's strange.
- Use chain for map too:
```javascript
_.chain($jsFollowEls)
 .map(...)
```
In fact I find the whole mapObject/pluck stuff impossible to read. If I understand it well, would not it be better to write it this way?
```javascript
const records = {};
for (const el of $jsFollowEls) {
    const model = el.dataset.object;
    if (!(model in records)) {
        records[model] = [];
    }
    records[model].push(parseInt(el.dataset.id));
}
```
It is probably 1000x faster too.

Reply:
Changed it.